File: android/resourceextensions/src/main/java/com/shopify/resourceextensions/Context.kt
package com.shopify.resourceextensions

class Context {

    operator fun get(value: String): Factory? {
        return when (value) {
            "#" -> PropertiesFactory()
            "label" -> LabelFactory()
            "button" -> ButtonFactory()
            "textfield" -> TextFieldFactory()
            "checkbox" -> CheckBoxFactory()
            else -> null
        }
    }

}

File: ios/AppExtensionLib/AppExtensionLib/Context.swift
//
//  Context.swift
//  AppExtensionLib
//
//  Created by Dave Pelletier on 2019-10-29.
//  Copyright © 2019 Shopify. All rights reserved.
//

import Foundation

public class Context {
    
    func get(value: String) -> Factory? {
        switch value {
        case "#":
            return PropertiesFactory()
        case "label":
            return LabelFactory()
        default:
            return nil
        }
    }
    
}
